docs/adr/adr-11-blocksync-overhaul.md

```diff
@@ -0,0 +1,329 @@
+```
+
+##### `share.EDSStore.GetCAR`
+
+To read an EDS as a byte stream `GetCAR` method is introduced. Internally it
+
+- Converts `DataRoot` into the [`shard.Key`](<https://github.com/filecoin-project/dagstore/blob/master/shard/key.go#L12>
+- Gets Mount by `shard.Key` from [`mount.Registry`](https://github.com/filecoin-project/dagstore/blob/master/mount/registry.go#L22)
+- Returns `io.ReadeCloser` from [`Mount.Fetch`](https://github.com/filecoin-project/dagstore/blob/master/mount/mount.go#L71)
+
+NOTE: It might be necessary to acquire EDS mount via `DAGStore` keeping `ShardAccessor`
+and closing it when operation is done. This has to be confirmed.
+
+```go
+// GetCAR takes DataRoot and returns a buffered reader to respective EDS serialized as CARv1 file.
+// 
+// The Reader strictly reads the first quadrant(1/4) of EDS omitting all the NMT Merkle proofs.
+// Integrity of the store data is not verified. 
+// 
+// Caller must Close returned reader after reading.
+func (s *Store) GetCAR(context.Context, DataRoot) (io.ReadCloser, error)
+```
+
+##### `share.EDSStore.Blockstore`
+
+`Blockstore` method return a `Blockstore` interface implementation instance, providing random access over share and NMT
+Merkle proof in every stored EDS. It is required for FNs/BNs to serve DAS requests over the Bitswap and for reading data
+by namespace. // TODO Link to the section
+
+There is a [frozen/un-merged implementation](https://github.com/filecoin-project/dagstore/pull/116) of `Blockstore`
+over `DAGStore` and CARv2 indexes.
+
+NOTE: We can either use it(and finish if something is missing for our case) or implement custom optimized for our needs.
+
+```go
+// Blockstore returns an IPFS Blockstore providing access to individual shares/nodes of all EDS 
+// registered on the Store. NOTE: The Blockstore does not store full Celestia Blocks, but IPFS blocks. 
+// We represent `shares` and NMT Merkle proofs as IPFS blocks and IPLD nodes, so that Bitswap can access those.
+func (s *Store) Blockstore() blockstore.Blockstore
+
+```
+
+##### `share.EDSStore.Get`
+
+To read an entire EDS `Get` method is introduced. Internally it:
+
+- Gets serialized EDS `io.Reader` via `Store.GetCAR`
+- Deserializes EDS and validates it via `share.ReadEDS`
+
+NOTE: It's not necessary, but an API ergonomics/symmetry nice-to-have
+
+```go
+// Get reads EDS out of Store by given DataRoot.
+// 
+// It reads only one quadrant(1/4) of the EDS and verifies integrity of the stored data by recomputing it.
+func (s *Store) Get(context.Context, DataRoot) (*rsmt2d.ExtendedDataSquare, error)
+
+```
+
+##### `share.EDSStore.Remove`
+
+To remove stored EDS `Remove` methods is introduced. Internally it:
+
+- Converts `DataRoot` into the [`shard.Key`](<https://github.com/filecoin-project/dagstore/blob/master/shard/key.go#L12>
+- Removes respective `Mount` and `Shard` from `DAGStore`
+  - `DAGStore` cleans up indices and `mount.Registery` itself
```

Comment:
doesn't need to do anything on the registry, or remove any mount

Reply:
Related to the discussions re getting car Reader above

Reply:
still nothing to clean on the registry

Reply:
Corrected

Reply:
There is still no mount to remove from the DAGStore, only the Shard. The mount is not stored in the DAGStore/Registry (but the template, which is used to create ALL mounts, is)

Reply:
The mount is [part](https://github.com/filecoin-project/dagstore/blob/12408ab3874cd3be952e207d00627253d9bb4878/shard.go#L40) of the Shard, though. I will clarify the removal still

Reply:
The mount is [part](https://github.com/filecoin-project/dagstore/blob/12408ab3874cd3be952e207d00627253d9bb4878/shard.go#L40) of the Shard. The mount's path also persisted. We still have to mention Mounts because while being different from Shard, we still create those and hence should remove those as well. 
I will mention that the Mount is cleared up on Shard destruction and put more emphasis on the Shard.

Reply:
Clarified